torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -156,26 +156,31 @@ c10::intrusive_ptr<JitFuture> sendMessageWithAutograd(
       forceGradRecording,
       agent.getDeviceMap(dst));
 
-  c10::intrusive_ptr<JitFuture> fut;
   // If profiler is enabled, wrap this message with profiling metadata that will
   // tell the remote end to process this request with the profiler enabled.
-  if (!forceDisableProfiling && torch::autograd::profiler::profilerEnabled()) {
-    auto profilerConfig = torch::autograd::profiler::getProfilerConfig();
-    TORCH_CHECK(
-        torch::profiler::impl::profilerType() ==
-            torch::profiler::impl::ActiveProfilerType::LEGACY,
-        "Currently only the legacy autograd profiler is supported in a distributed context.");
-    auto msgWithProfiling = getMessageWithProfiling(
-        std::move(msg),
-        rpc::MessageType::RUN_WITH_PROFILING_REQ,
-        // NOLINTNEXTLINE(performance-move-const-arg)
-        std::move(profilerConfig));
-    fut = agent.send(dst, std::move(msgWithProfiling), rpcTimeoutSeconds);
-  } else {
-    fut = agent.send(dst, std::move(msg), rpcTimeoutSeconds);
+  if (!forceDisableProfiling) {
+    switch (torch::profiler::impl::profilerType()) {
+      case torch::profiler::impl::ActiveProfilerType::LEGACY:
+        {
+          auto profilerConfig = torch::autograd::profiler::getProfilerConfig();
+          auto msgWithProfiling = getMessageWithProfiling(
+            std::move(msg),
+            rpc::MessageType::RUN_WITH_PROFILING_REQ,
+            // NOLINTNEXTLINE(performance-move-const-arg)
+            std::move(profilerConfig));
+          return agent.send(dst, std::move(msgWithProfiling), rpcTimeoutSeconds);
+        }
+      case torch::profiler::impl::ActiveProfilerType::KINETO:
+        TORCH_WARN_ONCE(
+          "Profiling a distributed call with the Kineto profiler will profile "
+          "the caller, but not the worker.");
```

Comment:
Do we have a github issue tracking this so that we can enable this functionality eventually? Also, what is the reason that this functionality depends on the profiler type? Intuitively it seems like remote profiling should be an RPC thing and not specific to a particular type of profiler.

Reply:
No issue that I'm aware of.

There's no reason that RPC couldn't support both cases, it just hasn't been implemented yet. Basically RPC has to know which profiler to instantiate on the remote, and you have to have different code to plumb back to the caller. (Autograd and Kineto are distinct profilers from an API perspective and the implementation now just reflects that.)